File: Dddml.Wms.HttpServices.JavaClient/src/org/dddml/wms/domain/RolePermissionStateEventIdFlattenedDto.java
package org.dddml.wms.domain;

import java.util.HashMap;
import java.util.Map;

public class RolePermissionStateEventIdFlattenedDto
{


    public static final String[] propertyNames = new String[]{
            "idRoleId",
            "idPermissionId",
            "version",
        };

    public static final String[] propertyTypes = new String[]{
            "String",
            "String",
            "Long",
        };

    public static final Map<String, String> propertyTypeMap;

    static {
        propertyTypeMap = new HashMap<String, String>();
        for (int i = 0; i < propertyNames.length; i++ ) {
            propertyTypeMap.put(propertyNames[i], propertyTypes[i]);
        }
    }
	
    private RolePermissionStateEventId value;

    public RolePermissionStateEventIdFlattenedDto()
    {
        this(new RolePermissionStateEventId());
    }

    public RolePermissionStateEventIdFlattenedDto(RolePermissionStateEventId value)
    {
        this.value = value;
    }

    public RolePermissionStateEventId toRolePermissionStateEventId()
    {
        return this.value;
    }


    public String getIdRoleId()
    {
        return this.value.getId().getRoleId();
    }

    public void setIdRoleId(String idRoleId)
    {
        this.value.getId().setRoleId(idRoleId);
    }

    public String getIdPermissionId()
    {
        return this.value.getId().getPermissionId();
    }

    public void setIdPermissionId(String idPermissionId)
    {
        this.value.getId().setPermissionId(idPermissionId);
    }

    public Long getVersion()
    {
        return this.value.getVersion();
    }

    public void setVersion(Long version)
    {
        this.value.setVersion(version);
    }


    @Override
    public boolean equals(Object obj)
    {
        if (obj == this) {
            return true;
        }
        if (obj == null || obj.getClass() != this.getClass()) {
            return false;
        }

        RolePermissionStateEventIdFlattenedDto other = (RolePermissionStateEventIdFlattenedDto)obj;
        return value.equals(other.value);
    }

    @Override
    public int hashCode()
    {
        return value.hashCode();
    }

}

File: Dddml.Wms.HttpServices.PhpClient/Dddml/Wms/Domain/RolePermissionStateEventIdFlattenedDto.php
<?php

namespace Dddml\Wms\Domain;

use JMS\Serializer\Annotation\Type;
use Dddml\Serializer\Type\Long;
use Dddml\FlattenedDtoInterface;

class RolePermissionStateEventIdFlattenedDto implements FlattenedDtoInterface
{

    const PROPERTIES = [
            'idRoleId' => 'string',
            'idPermissionId' => 'string',
            'version' => 'Long',
        ];

    public static function getPropertyNames()
    {
        return array_keys(static::PROPERTIES);
    }

    public static function getPropertyTypes()
    {
        return array_values(static::PROPERTIES);
    }

	
    /**
     * @var RolePermissionStateEventId
     */
    private $value;

    /**
     * @param RolePermissionStateEventId $value
     */
    public function __construct(RolePermissionStateEventId $value = null)
    {
        if ($value) {
            $this->value = $value;
        } else {
            $this->value = new RolePermissionStateEventId();
        }
    }

    /**
     * @return RolePermissionStateEventId
     */
    public function toRolePermissionStateEventId()
    {
        return $this->value;
    }

    /**
     * @return string
     */
    public function getIdRoleId()
    {
        return $this->value->getId()->getRoleId();
    }

    /**
     * @param string $idRoleId
     */
    public function setIdRoleId($idRoleId)
    {
        $this->value->getId()->setRoleId($idRoleId);
    }

    /**
     * @return string
     */
    public function getIdPermissionId()
    {
        return $this->value->getId()->getPermissionId();
    }

    /**
     * @param string $idPermissionId
     */
    public function setIdPermissionId($idPermissionId)
    {
        $this->value->getId()->setPermissionId($idPermissionId);
    }

    /**
     * @return Long
     */
    public function getVersion()
    {
        return $this->value->getVersion();
    }

    /**
     * @param Long $version
     */
    public function setVersion($version)
    {
        $this->value->setVersion($version);
    }

    /**
     * @return string
     */
    public function toString()
    {
        $pValues = [
            $this->getIdRoleId(),
            $this->getIdPermissionId(),
            $this->getVersion(),
        ];
        return implode(',', $pValues);
    }

    /**
     * @param string $data
     *
     * @return RolePermissionStateEventIdFlattenedDto
     */
    public function fromString($data)
    {
        $pValues = explode(',', $data);
        $this->setIdRoleId($pValues[0]);
        $this->setIdPermissionId($pValues[1]);
        $this->setVersion($pValues[2]);
        return $this;
    }

}

